### PR TITLE
#204396,204397,204394,204393,204390, : Camp dashboard and flow.

### PR DESCRIPTION
**Pull request by swapnilphalke03**
### I have ensured that following `Pull Request Checklist` is taken care of before sending this PR
* [ ] Code is formatted as per format decided
* [ ] Updated acceptance criteria in tracker
* [ ] Updated test cases in test-cases-tracker
https://tracker.tekdi.net/issues/204396
https://tracker.tekdi.net/issues/204397
https://tracker.tekdi.net/issues/204394
https://tracker.tekdi.net/issues/204393

### CODE REVIEW

File: apps/front-end/src/pages/front-end/BenificiaryEnrollment.js [hunk 1]
```diff
@@ -69,6 +69,7 @@ export default function BenificiaryEnrollment() {
           {[
             "identified",
             "ready_to_enroll",
+            "applied_but_pending",
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/BenificiaryEnrollment.js:72]: Remove

**Review by sagarkoshti1990 — CHANGES_REQUESTED** [review]

File: apps/front-end/src/component/BaseInput.js [hunk 2]
```diff
@@ -575,6 +576,36 @@ export const MultiCheck = ({
   );
 };
 
+const CheckUncheck = ({ required, schema, value, onChange }) => {
```
Comment (manojLondhe) [inline apps/front-end/src/component/BaseInput.js:579]: add comment for what is this for?

File: apps/front-end/src/component/formCustomeInputs/FileUpload.js [hunk 3]
```diff
@@ -24,7 +24,7 @@ const FileUpload = ({ options, value, onChange, required, schema }) => {
     const item = {
       file,
       document_type,
-      user_id: userId,
+      user_id: userId || localStorage.getItem("id"),
```
Comment (manojLondhe) [inline apps/front-end/src/component/formCustomeInputs/FileUpload.js:27]: what does localStorage.getItem("id") contain?

File: lib/common-lib/src/services/campRegistoryService.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
+import { post, patch, handleResponseException } from './RestClient'
```
Comment (manojLondhe) [inline lib/common-lib/src/services/campRegistoryService.js:1]: just rename as campService.js

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: apps/front-end/src/pages/front-end/Camp/CampDashboard.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        </Alert>
+      </Box>
+      <Box padding="6">
+        <AdminTypo.H2>{t("HOW_TO_START_CAMP")}</AdminTypo.H2>
+        <HStack mt={2}>
+          <iframe
+            width="100%"
+            height="315"
+            src="https://www.youtube.com/embed/_Tbo0cATRGM?si=G7KEFIkHPLg1mFQy"
+            title="YouTube video player"
```
[the remark this answers is not in the document]
Comment (swapnilphalke03) [reply apps/front-end/src/pages/front-end/Camp/CampDashboard.js]: Resolved.

File: apps/front-end/src/routes/routes.js [hunk 6]
```diff
@@ -220,5 +235,23 @@ export default [
...
+  {
+    path: "/camps/:id",
+    component: CampRegistration,
+  },
+  {
+    path: "/camps/:id/:step",
+    component: CampForm,
+  },
+  {
+    path: "/camps/LearnerList",
```
[the remark this answers is not in the document]
Comment (swapnilphalke03) [reply apps/front-end/src/routes/routes.js]: Change done.

File: lib/common-lib/src/services/campRegistoryService.js [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+      return result?.data
+    } else {
+      return {}
+    }
+  } catch (e) {
+    return handleResponseException(e)
+  }
+}
+
+export const createConsent = async (params = {}, header = {}) => {
```
[the remark this answers is not in the document]
Comment (swapnilphalke03) [reply lib/common-lib/src/services/campRegistoryService.js]: Resolved

File: lib/common-lib/src/services/campRegistoryService.js (hunk 4 above)
Comment (swapnilphalke03) [reply lib/common-lib/src/services/campRegistoryService.js]: Renamed.

File: apps/front-end/src/component/BaseInput.js (hunk 2 above)
Comment (swapnilphalke03) [reply apps/front-end/src/component/BaseInput.js:580]: comment added

File: apps/front-end/src/component/formCustomeInputs/FileUpload.js (hunk 3 above)
Comment (swapnilphalke03) [reply apps/front-end/src/component/formCustomeInputs/FileUpload.js]: id of the logged-in user, comment added

**Pull request merged**